### DISCUSSION TITLE
Architecture for inheriting FelizProps.props

### DISCUSSION THREAD

**Discussion by Wouter8** [sydsutton/FS.FluentUI#6] [Q&A] [answered]
Hi, thanks for this extensive library.

I'm wondering what the idea is behind inheriting the `FelizProps.props` type. I understand that all components can use their specific type for all props (`[component].[prop]`), even if it is a prop that would normally be applied through `prop.[prop]`.

I thought the idea was that components that inherit this type would have a container to which all these normal HTML props are applied. Is this true?

When I picked a random component to check this reasoning I couldn't find this in the React code.
[Here](https://github.com/microsoft/fluent-ui-react/blob/master/packages/react/src/components/MenuButton/MenuButton.tsx#L29C10-L29C10) you can find the prop type for MenuButton in the React library, [here](https://github.com/sydsutton/FS.FluentUI/blob/master/src/FS.FluentUI/Props.fs#L925) is the prop type in your library. The only default HTML prop there is `className`. Do you inherit `FelizProps.props` just for the `className` prop?

I'm looking at your library to decide how I want to structure a bindings library of my own for MantineUI :)
Thanks in advance.

**Comment by sydsutton** [comment, marked as answer]
Yeah, it's because I wanted all eg- Dropdown properties to be exclusively IDropdownProp's. When i want to do `dropdown.ariaLabeledBy "someLabel"`, I don't have to explicitly add the ariaLabeledBy property under the Dropdown props,-- I can just pass the type that I want to the reusable FelizProps and it stays type-safe. Somebody opened an issue a few months ago suggesting another way of implementing the same idea, though, so I would look at that in order to make an informed decision. But FelizProps equate to ComponentProps in version 9 of FluentUI code, not version 8. Some components have them, some don't. I hope this helps!

**Reply by Wouter8** [reply]
Where can I find this issue/discussion?

**Reply by Wouter8** [reply]
Got it, it was a PR (#5). Thanks!

**Reply by sydsutton** [reply]
OK cool, and not a problem! Good luck!
